backend/main-service/src/main/java/kkakka/mainservice/cart/application/CartService.java

```diff
@@ -29,32 +33,60 @@ public CartService(CartRepository cartRepository, CartItemRepository cartItemRep
         this.memberRepository = memberRepository;
     }
 
-    public String saveCartItem(CartRequestDto cartRequestDto) {
+    @Transactional
+    public Long saveOrUpdateCartItem(CartRequestDto cartRequestDto) {
 
         /* 테스트 데이터 */
         Optional<Member> member = memberRepository.findById(cartRequestDto.getMemberId());
 
-        /* 로그인 상태 체크 */
+        Cart cart = findOrCreateCart(member.get());
+        Optional<CartItem> cartItem = Optional.ofNullable(
+                cartItemRepository.findByMemberIdandProductId(member.get().getId(), cartRequestDto.getProductId())
+        );
 
-        /* 장바구니 존재 유무 체크 ex) 장바구니에 담아둔 아이템 있는지 */
-        Cart cart = new Cart();
-        cart = cartRepository.findByMemberId(cartRequestDto.getMemberId());
-        if (cart == null) {
-            cartRepository.save(new Cart(member.get()));
-            cart = cartRepository.findByMemberId(cartRequestDto.getMemberId());
+        // 동일 상품 이미 있으면 수량 업데이트
+        if (cartItem.isPresent()) {
+            Integer preQuantiry = cartItem.get().getQuantity();
+            cartItem.get().setQuantity(preQuantiry + cartRequestDto.getQuantity());
+            cartItemRepository.save(cartItem.get());
+            return member.get().getId();
         }
 
-        /* 현재 장바구니에 동일한 상품 있는지 체크 */
-        CartItem item = cartItemRepository.findByMemberIdandProductId(member.get().getId(), cartRequestDto.getProductId()); // Test용 Member
-//        System.out.println(item.toString());
-
         /* 장바구니 아이템 추가 */
         Optional<Product> product = productRepository.findById(cartRequestDto.getProductId());
-        cartItemRepository.save(new CartItem(cart, product.get(), cartRequestDto.getQuantity()));
+        cartItem = Optional.ofNullable(new CartItem(cart, product.get(), cartRequestDto.getQuantity()));
+        cartItemRepository.save(cartItem.get());
+
+        return member.get().getId();
+    }
+
+    /* 멤버 장바구니 목록 조회 */
+    public List<CartResponseDto> findAllCartItemByMember(Member member) {
+
+        List<CartItem> cartItemList = cartItemRepository.findAllByMemberId(member.getId());
+
+        List<CartResponseDto> result = new ArrayList<>();
+        cartItemList.forEach(c -> {
+            result.add(CartResponseDto.from(c));
+        });
```

Comment:
스트림을 활용해보아요😉 저라면 바로 리턴할 것 같아요.

```suggestion
        List<CartItem> cartItemList = cartItemRepository.findAllByMemberId(member.getId());
        
        return cartItemList.stream()
                      .map(cartResponseDto::from)
                      .collect(Collectors.toList());
```

Reply:
for 문을 사용하면 확실히 누구나 알아볼 수 있는 가독성이 생기죠.
하지만 스트림도 잘 활용한다면 충분히 가독성 좋고 단순한 코드를 만들 수 있다고 생각합니다!ㅎㅎ

Reply:
Optional 도 그렇고 스트림도 그렇고 잘 알면 가독성도 좋고 코드도 더 개발자스럽고 좋아보여요
예시가 있으면 활용할 수는 있겠는데 아직 머리속에 자리잡지 않아서 바로바로 안떠오르네요
이것도 반영하도록 하겠습니다 ~! 감사